gsoc-2025/ideas.md

```diff
@@ -162,3 +162,146 @@ Finally, at all steps tests should be implemented that compare original function
 * https://github.com/dpeckett/archivefs/tree/main/erofs
 
 ---
+
+### Testing Framework for Unikraft Builds
+
+| | |
+| **Difficulty** | 3/5 |
+| **Project Size** | Variable (175 or 350 hours) |
+| **Maximum instances** | 1 |
+| **Constraints/requirements** | Python knowledge, Linux CLI |
+
+#### Description
+
+We are currently developing a [testing framework](https://github.com/unikraft-upb/catalog/tree/razvand/generator/new-design/utils/new-design) that is able to multiplex the variety of configuration options, VMMs, hypervisors, architectures, boot protocols, to validate the successful building and running of unikernel images.
+This framework is able to configure, build, run and test the variety of Unikraft builds.
+It is written in Python and is subject to improvements and refactoring.
+
+We are looking to augment the testing infrastructure to make it seamless to be used by Unikraft developers and users.
+To this end we aim to:
+
+- Consolidate the testing framework as a separate project inside its own repository.
+- Have the testing framework work out-of-the-box with the [`catalog`](https://github.com/unikraft/catalog) and [`catalog-core`](https://github.com/unikraft/catalog-core) repositories.
+- Integrate the testing framework with the CI/CD system used in the [Unikraft organization repositories](https://github.com/unikraft) to automatically validate builds for contributions.
+
+#### Reading & Related Material
+
+* https://github.com/unikraft/catalog
+* https://github.com/unikraft/catalog-core
+* https://github.com/unikraft-upb/catalog/tree/razvand/generator/new-design/utils/new-design
+
+---
```

Comment:
I do not understand if this separator should be for every entry, or only at the end

Can you go around and do it correctly everywhere? (add everywhere, or add only at the end)